### PR TITLE
Created dock widget for the plug-in

### PR DESCRIPTION
**Pull request by GeOdin**
Created a dock widget for the plug-in (instead of the current dialog window), 
to improve the overview of the plug-in and to improve the ease of implementation of future features.

### CODE REVIEW

File: lizard_downloader.py [hunk 1]
```diff
@@ -44,16 +45,16 @@ def __init__(self, iface):
...
                 QCoreApplication.installTranslator(self.translator)
 
-        # Create the dialog (after translation) and keep reference
-        self.dlg = LizardDownloaderDialog()
-
         # Declare instance attributes
         self.actions = []
         self.menu = self.tr(u'&Lizard Viewer')
         # TODO: We are going to let the user set this up in a future iteration
         self.toolbar = self.iface.addToolBar(u'LizardDownloader')
```
Comment (jackieleng) [inline lizard_downloader.py:52]: Shouldn't this be changed to lizard viewer?

File: lizard_downloader.py [hunk 2]
```diff
@@ -4,12 +4,13 @@
...
 from PyQt4.QtCore import QCoreApplication
 from PyQt4.QtCore import QSettings
+from PyQt4.QtCore import Qt
 from PyQt4.QtCore import QTranslator
 from PyQt4.QtCore import qVersion
 from PyQt4.QtGui import QAction
 from PyQt4.QtGui import QIcon
 
-from lizard_downloader_dialog import LizardDownloaderDialog
+from dockwidget import Ui_DockWidget
```
Comment (jackieleng) [inline lizard_downloader.py]: Ambiguous import, use:

``from .dockwidget import  .....``

**Review by jackieleng — COMMENTED** [review]
Looks good, but try not to do too many things in one PR. (in this case, adding a submodule and creating a dockwidget). Will test it next week.

File: lizard_downloader.py (hunk 2 above)
Comment (GeOdin) [reply lizard_downloader.py]: Thanks.

File: lizard_downloader.py (hunk 1 above)
Comment (GeOdin) [reply lizard_downloader.py:52]: Reference to dialog no longer needed with dockwidget.

Comment (jackieleng) [reply lizard_downloader.py:52]: nvm, can be done in another PR

File: dockwidget.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import os
+
+from PyQt4 import QtGui, uic
+from PyQt4.QtCore import pyqtSignal
+
+FORM_CLASS, _ = uic.loadUiType(os.path.join(
+    os.path.dirname(__file__), 'dockwidget.ui'))
+
+
+class Ui_DockWidget(QtGui.QDockWidget, FORM_CLASS):
```
Comment (jackieleng) [inline dockwidget.py:13]: ``Ui_DockWidget`` is probably a QGIS default name? I think a more descriptive name would make this much better.

Comment (jackieleng) [reply dockwidget.py:13]: We can do this in another PR, I'll merge it for now.

**Pull request merged**